packages/spiderable/spiderable_client.js

```diff
@@ -17,14 +17,52 @@
 
 Spiderable._initialSubscriptionsStarted = false;
 
+Spiderable._onReadyHook = new Hook({
+  debugPrintExceptions: "onReadyHook callback"
+});
+
+// register a new onReady hook for validation
+Spiderable.onReady = function (fn) {
```

Comment:
I have no preferences on what the hook or registering function are named, please just provide me which names you'd like to use for each.